sensors/test_deps.go

```diff
@@ -41,6 +44,17 @@ func SetupDeps(lidarName string) resource.Dependencies {
 		deps[camera.Named(lidarName)] = getFinishedReplayLidar()
 	}
 
+	switch imuName {
+	case "good_imu":
+		deps[movementsensor.Named(imuName)] = getGoodIMU()
+	case "invalid_imu":
+		deps[movementsensor.Named(imuName)] = getInvalidIMU()
```

Comment:
Might be worth splitting of a second invalid case. One being when the properties are valid but the functions are not, and another whether both the properties and functions are invalid.

Reply:
I feel like the "bad_imu" essentially covers this, since as soon as an IMU that does not have the required properties is detected, we get an error. See test on line 130 of sensors_test.go

Reply:
You're right it does handle it, thanks for checking. I would suggest switching their names as invalid imu more correctly correlates to an imu that does not have the valid properties for the operation

Reply:
Communicated offline - will leave as is for now to match lidar convention